tests/integration_tests/test_cli.py

```diff
@@ -300,3 +301,86 @@ def test_preprocess_cli(csv_filename):
         config_filename = os.path.join(tmpdir, "config.yaml")
         dataset_filename = _prepare_data(csv_filename, config_filename)
         _run_ludwig("preprocess", dataset=dataset_filename, preprocessing_config=config_filename)
+
+
+@pytest.mark.distributed
+@pytest.mark.parametrize("second_seed_offset", [0, 1])
+@pytest.mark.parametrize("random_seed", [1919, 31])
+@pytest.mark.parametrize("type_of_run", ["train", "experiment"])
+@pytest.mark.parametrize("backend", ["local", "horovod"])
+def test_reproducible_cli_runs(
+    backend: str, type_of_run: str, csv_filename: str, random_seed: int, second_seed_offset: int
+) -> None:
+    """
+    Test for reproducible training using `ludwig experiment|train --dataset`.
+    Args:
+        backend (str): backend to use
+        type_of_run(str): type of run, either train or experiment
+        csv_filename(str): file path of dataset to use
+        random_seed(int): random seed integer to use for test
+        second_seed_offset(int): zero to use same random seed for second test, non-zero to use a different
+            seed for the second run.
+
+    Returns: None
+    """
+    with tempfile.TemporaryDirectory() as tmpdir:
+        config_filename = os.path.join(tmpdir, "config.yaml")
+        dataset_filename = _prepare_data(csv_filename, config_filename)
+
+        if backend == "local":
+            command_to_run = _run_ludwig
+        else:
+            command_to_run = _run_ludwig_horovod
+
+        # run first model
+        command_to_run(
+            type_of_run,
+            dataset=dataset_filename,
+            config=config_filename,
+            output_directory=tmpdir,
+            skip_save_processed_input="",  # skip saving preprocessed inputs for reproducibility
```

Comment:
This should be `False`, right?

Reply:
Given the way the function that generates the cli command, we only need `--skip_save_preprocessed_input`, which is assumed to be true.  For cli this is not a keyword parameter, like the api version. 

Based on my testing, I believe this is correct.

Reply:
@justinxzhao  a bit more specifics on why `skip_save_processed_input=""` is used.

From a cli perspective the argument is defined as follows:
```
    parser.add_argument(
        "-sspi",
        "--skip_save_processed_input",
        help="skips saving intermediate HDF5 and JSON files",
        action="store_true",
        default=False,
```

The presence or absence of the argument on the command line determines `True` or `False`, respectively.

Second in `test_cli.py`, this function generates the cli arguments and runs the cli command
```
def _run_commands(commands, **ludwig_kwargs):
    for arg_name, value in ludwig_kwargs.items():
        commands += ["--" + arg_name, value]
    cmdline = " ".join(commands)
    print(cmdline)
    completed_process = subprocess.run(cmdline, shell=True, stdout=subprocess.PIPE, env=os.environ.copy())
    assert completed_process.returncode == 0


    return completed_process
```

The relevant code is `command += ["--" + arg_name, value]` is always looking for a value.  So by using `skip_save_processed_input=""`, the `_run_commands()` function generates this cli argument:
 ```
--skip_save_processed_input
```

Does this answer the question?